### ISSUE TITLE
Clarify "study type and dependencies" behavior

### ISSUE THREAD

**Issue by kimberscott** [lookit/lookit-api#390]
**TL;DR**: The "Study type and dependencies" section is confusing and leads to unnecessary extra builds.

**Narrative**
"As a researcher, I want to be confident about what pressing the 'save' button on my study will do (including invalidating existing build). I do not want to have to re-build dependencies if I haven't changed anything. I am generally a bit confused about the study type and dependencies section presently."

**Acceptance Criteria**

- [x] Fix link to docs in study dependencies section and clarify text
- [x] Call the JSON something clear that people can refer to so they stop calling it the "code for their study"; call study dependencies & building study dependencies something clear as well 
- [x] Collapse study edit + study type/dependencies edit fields into one form served by the same 'save' button so people stop pressing the save button at the bottom mistakenly.
- [x] Saving only invalidates existing build if the commit SHA changes.
- [x] Saving generates additional warning if commit SHA is changing & invalidating a build.
- [x] Give `player_repo_url` and `last_known_player_sha` friendly display names.
- [x] Don't leave JSON expanded after editing it, so the rest of the form is more visible
- [x] Flag is_previewing, is_building. Don't allow re-building under these circumstances. Remove flags when invalidating builds and when done
- [x] Warning displayed if researcher clicks "build dependencies / build preview dependencies" from study detail or edit views and build already exists OR is in progress.
- [x] Make sure study create view works as well as study edit
- [x] Commit message & date of commit SHA displayed next to the last_known_player_sha field
- [x] "Check for updates" button near commit SHA field added. Displays either "Your code is up-to-date" or "There is an update available: [commit SHA], [commit message], [date]. You are currently using: [commit SHA], [commit message], [date]."
- [x] Update lookit-docs accordingly - main & tutorial. (replace everything to do with: dependencies, study type, last_known_player_sha, etc; update screenshots)

**Implementation Notes**
